benchmarks/linear_programming/run_mps_files.sh

```diff
@@ -0,0 +1,276 @@
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# This script runs MPS (Mathematical Programming System) files in parallel across multiple GPUs
+# It supports running linear programming (LP) and mixed-integer programming (MIP) problems
+#
+# Usage:
+#   ./run_mps_files.sh --path /path/to/mps/files --ngpus 4 --time-limit 3600 --output-dir results
+#
+# Arguments:
+#   --path        : Directory containing .mps files (required)
+#   --ngpus       : Number of GPUs to use (default: 1)
+#   --time-limit  : Time limit in seconds for each problem (default: 360)
+#   --output-dir  : Directory to store output log files (default: current directory)
+#   --relaxation  : Run relaxation instead of solving the MIP
+#   --mip-heuristics-only : Run mip heuristics only
+#   --write-log-file : Write log file
+#   --num-cpu-threads : Number of CPU threads to use
+#   --batch-num : Batch number
+#   --n-batches : Number of batches
+#   --log-to-console : Log to console
+#
+# Examples:
+#   # Run all MPS files in /data/lp using 2 GPUs with 1 hour time limit
+#   ./run_mps_files.sh --path /data/lp --ngpus 2 --time-limit 3600
+#
+#   # Run with specific GPU devices using CUDA_VISIBLE_DEVICES
+#   CUDA_VISIBLE_DEVICES=0,2,3 ./run_mps_files.sh --path /data/mip
+#
+#   # Run with custom output directory
+#   ./run_mps_files.sh --path /data/lp --output-dir ~/results/lp_benchmark
+#
+# Notes:
+#   - Files are distributed dynamically across available GPUs for load balancing
+#   - Each problem's results are logged to a separate file in the output directory
+#   - The script uses file locking to safely distribute work across parallel processes
+#   - If CUDA_VISIBLE_DEVICES is set, it can override the --ngpus argument
+
+# Help function
+print_help() {
+    cat << EOF
+This script runs MPS (Mathematical Programming System) files in parallel across multiple GPUs.
+It supports running linear programming (LP) and mixed-integer programming (MIP) problems.
+
+Usage:
+    $(basename "$0") --path /path/to/mps/files [options]
+
+Required Arguments:
+    --path PATH          Directory containing .mps files
+
+Optional Arguments:
+    --ngpus N           Number of GPUs to use (default: 1)
+    --time-limit N      Time limit in seconds for each problem (default: 360)
+    --output-dir PATH   Directory to store output log files (default: current directory)
+    --relaxation       Run relaxation instead of solving the MIP
+    --mip-heuristics-only  Run mip heuristics only
+    --write-log-file   Write log file
+    --num-cpu-threads  Number of CPU threads to use
+    --batch-num        Batch number
```

Comment:
May be more details on batch number and n-batches and does it support LP or MIP or both.

Reply:
Do we need to expand more on these options ?